src/pudl/transform/eia861.py

```diff
@@ -176,14 +434,386 @@
+
+
+def _clean_nerc(df, idx_cols):
+    """Clean NERC region entries and make new rows for multiple nercs.
+
+    This function examines reported NERC regions and makes sure the output column of
+    the same name has reliable, singular NERC region acronyms. To do so, this function
+    identifies entries where there are two or more NERC regions specified in a single cell
+    (such as SPP & ERCOT) and makes new, duplicate rows for each NERC region. It also
+    converts non-recognized reported nerc regions to 'UNK'.
+
+    Args:
+        df (pandas.DataFrame): A DataFrame with the column 'nerc_region' to be
+            cleaned.
+        idx_cols (list): A list of the primary keys.
+
+    Returns:
+        pandas.DataFrame: A DataFrame with correct and clean nerc regions.
+
+    """
+    idx_no_nerc = idx_cols.copy()
+    if 'nerc_region' in idx_no_nerc:
+        idx_no_nerc.remove('nerc_region')
+
+    # Split raw df into primary keys plus nerc region and other value cols
+    nerc_df = df[idx_cols].copy()
+    other_df = df.drop('nerc_region', axis=1).set_index(idx_no_nerc)
+
+    # Make all values upper-case
+    # Replace all NA values with UNK
+    # Make nerc values into lists to see how many separate values are stuffed into one row (ex: 'SPP & ERCOT' --> ['SPP', 'ERCOT'])
+    nerc_df = (
+        nerc_df.assign(
+            nerc_region=(lambda x: (
+                x.nerc_region
+                .str.upper()
+                .fillna('UNK')
+                .str.findall(r'[A-Z]+')))
+        )
+    )
+
+    # Record a list of the reported nerc regions not included in the recognized regions list (these eventually become UNK)
+    nerc_col = nerc_df['nerc_region'].tolist()
+    nerc_list = list(set([item for sublist in nerc_col for item in sublist]))
+    non_nerc_list = [
+        nerc_entity for nerc_entity in nerc_list if nerc_entity not in pc.RECOGNIZED_NERC_REGIONS + list(NERC_SPELLCHECK.keys())]
+    print(
+        f'The following reported NERC regions are not currently recognized and become UNK values: {non_nerc_list}')
+
+    # Function to turn instances of 'SPP_UNK' or 'SPP_SPP' into 'SPP'
+    def _remove_nerc_duplicates(entity_list):
+        if len(entity_list) > 1:
+            if 'UNK' in entity_list:
+                entity_list.remove('UNK')
+            if all(x == entity_list[0] for x in entity_list):
+                entity_list = [entity_list[0]]
+        return entity_list
+
+    # Go through the nerc regions, spellcheck errors, delete those that aren't recognized, and piece them back together
+    # (with _ separator if more than one recognized)
+    nerc_df['nerc_region'] = (
+        nerc_df['nerc_region']
+        .apply(lambda x: [i if i not in NERC_SPELLCHECK.keys() else NERC_SPELLCHECK[i] for i in x])
+        .apply(lambda x: sorted([i if i in pc.RECOGNIZED_NERC_REGIONS else 'UNK' for i in x]))
+        .apply(lambda x: _remove_nerc_duplicates(x))
```

Comment:
Codacy found an issue: [Lambda may not be necessary](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=6263807)